drivers/display/ek79652.c

```diff
@@ -0,0 +1,725 @@
+					config->lut_bw.len);
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	if (config->lut_wb.len) {
+		err = ek79652_write_cmd(dev, EK79652_CMD_LUTWB, config->lut_wb.data,
+					config->lut_wb.len);
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	if (config->lut_bb.len) {
+		err = ek79652_write_cmd(dev, EK79652_CMD_LUTBB, config->lut_bb.data,
+					config->lut_bb.len);
+		if (err < 0) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static int ek79652_update_partial_display(const struct device *dev, uint8_t *data, size_t len)
+{
+	int err;
+
+	LOG_DBG("Trigger partial update sequence");
+
+	err = ek79652_write_cmd(dev, EK79652_CMD_PDRF, data, len);
+	if (err < 0) {
+		return err;
+	}
+
+	ek79652_busy_wait(dev);
+
+	return 0;
+}
+
+static int ek79652_blanking_off(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int ek79652_blanking_on(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int ek79652_display_refresh(const struct device *dev, enum refresh_val type)
+{
+	struct ek79652_data *driver = dev->data;
+	int err;
+
+	if (type) {
+		ek79652_busy_wait(dev);
+
+		err = ek79652_update_display(dev);
+		if (err < 0) {
+			return err;
+		}
+		driver->full_refresh = true;
+
+	} else {
+		driver->initial = true;
+	}
+
+	driver->disp_refresh = true;
+
+	return 0;
+}
+
+static int ek79652_write(const struct device *dev, const uint16_t x, const uint16_t y,
+			const struct display_buffer_descriptor *desc,
+			const void *buf)
+{
+	const struct ek79652_config *config = dev->config;
+	struct ek79652_data *driver = dev->data;
+	uint16_t xs = (x % 8) ? x : (x - (x % 8));
+	uint16_t w = (desc->width % 8) ? desc->width : (desc->width - (desc->width % 8));
+	uint16_t h = desc->height;
+	size_t buf_len, buf_ptd_len;
+	int err;
+
+	LOG_DBG("x %u, y %u, height %u, width %u, pitch %u",
+		x, y, desc->height, desc->width, desc->pitch);
+
+	buf_len = MIN(desc->buf_size,
+		      desc->height * desc->width / EK79652_PIXELS_PER_BYTE);
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT(buf != NULL, "Buffer is not available");
+	__ASSERT(buf_len != 0U, "Buffer of length zero");
+	__ASSERT(!(desc->width % EK79652_PIXELS_PER_BYTE),
+		"Buffer width not multiple of %d", EK79652_PIXELS_PER_BYTE);
+
+	if ((h > config->height) || (w > config->width)) {
+		LOG_ERR("Position out of bounds");
+		return -EINVAL;
+	}
+
+	buf_ptd_len = buf_len + EK79652_PDT_REG_LENGTH;
+	uint8_t ptd[buf_ptd_len];
+
+	/* Setup Partial Window and enable Partial Mode */
+	sys_put_be16(xs, &ptd[EK79652_PDT_X_IDX]);
+	sys_put_be16(y, &ptd[EK79652_PDT_Y_IDX]);
+	sys_put_be16(w, &ptd[EK79652_PDT_W_IDX]);
+	sys_put_be16(h, &ptd[EK79652_PDT_H_IDX]);
+
+	memcpy((ptd + EK79652_PDT_REG_LENGTH), (uint8_t *)buf, buf_len);
+
+	err = ek79652_write_cmd(dev, EK79652_CMD_PTIN, NULL, 0);
+	if (err < 0) {
+		return err;
+	}
+
+	err = ek79652_write_cmd(dev, EK79652_CMD_PDTM2, ptd, buf_ptd_len);
+	if (err < 0) {
+		return err;
+	}
+
+	if (driver->disp_refresh == true) {
```

Comment:
What is the reason to call ek79652_write() with `driver->disp_refresh` set to false?